### PR TITLE
Add gesture handler

### PR DESCRIPTION
**Pull request by rachel-fenichel**
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/928)

### CODE REVIEW

File: core/gesture_handler.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The class that handlers gestures (drags, clicks, etc) across
```
Comment (picklesrus) [inline core/gesture_handler.js]: typo: handles

File: core/gesture_handler.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ * @namespace
+ */
+goog.provide('Blockly.GestureHandler');
+
+goog.require('Blockly.Gesture');
+goog.require('Blockly.Touch');
+
+
+/**
+ * Map from touch identifiers to in-process gestures.
```
Comment (picklesrus) [inline core/gesture_handler.js]: should this be in-progress?

File: core/gesture_handler.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ */
+'use strict';
+
+/**
+ * @name Blockly.GestureHandler
+ * @namespace
+ */
+goog.provide('Blockly.GestureHandler');
+
+goog.require('Blockly.Gesture');
```
Comment (picklesrus) [inline core/gesture_handler.js]: Does this already exist somewhere? I think you're just breaking up changes into pieces which is cool, but I want to make sure I'm not missing a piece myself :)

File: core/gesture_handler.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+Blockly.GestureHandler.MAX_GESTURES = 1;
+
+/**
+ * Look up the gesture that is tracking this touch stream.  May create a new
+ * gesture.
+ * @param {!Event} e Mouse event or touch event
+ * @return {Blockly.Gesture} The gesture that is tracking this touch stream,
+ *     or null if no valid gesture exists.
+ */
+Blockly.GestureHandler.gestureForEvent = function(e) {
```
Comment (picklesrus) [inline core/gesture_handler.js]: Are all of these methods intentionally static (not sure that's the right word in javascript)? In other words, is a GestureHandler not something you expect to have to construct and so anywhere in the code base or any developer can just call gestureForEvent, removeGestureForId, etc?

File: core/gesture_handler.js (hunk 3 above)
Comment (rachel-fenichel) [inline core/gesture_handler.js]: It doesn't exist yet except in my pending code.  Yes, I pulled this out from a larger change.

File: core/gesture_handler.js (hunk 4 above)
Comment (rachel-fenichel) [inline core/gesture_handler.js]: Correct

**Pull request merged**